vpr/src/place/place_constraints.h

```diff
@@ -34,9 +34,29 @@ bool cluster_floorplanning_legal(ClusterBlockId blk_id, const t_pl_loc& loc);
 bool is_macro_constrained(const t_pl_macro& pl_macro);
 
 /*
- * Returns region of valid locations for the head of the macro based on floorplan constraints
+ * Returns PartitionRegion for the head of the macro based on the floorplan constraints
+ * of all blocks in the macro.
  */
-PartitionRegion constrained_macro_locs(const t_pl_macro& pl_macro);
+PartitionRegion update_macro_head_pr(const t_pl_macro& pl_macro);
+
+/*
+ * Update the PartitionRegions of non-head members of a macro,
+ * based on the constraint that was calculated for the head region.
+ * The constraint that was calculated for the head region considered all
+ * constrained blocks in the macro, so to calculate the constraint for the rest
+ * of the blocks, it is just the head constraint with the offset of the member
+ * applied.
+ */
+PartitionRegion update_macro_member_pr(PartitionRegion& head_pr, const t_pl_offset& offset);
+
+/*
+ * Updates the floorplan constraints information for all constrained macros.
```

Comment:
Giving an example would help explain what this does (e.g. macro of length 2, both with floorplan constraints, implies a tighter floorplan constraint on the head block of the macro that we can compute by shifting and intersecting the per-member constraints).